### PR TITLE
Add Prometheus retention time period control

### PR DESCRIPTION
**Pull request by leifmadsen**

### CODE REVIEW

**Comment by leifmadsen** [comment]
## Testing and Results

Create `ServiceTelemetry` object with new retention period

```
$ oc get stf default -oyaml

apiVersion: infra.watch/v1beta1
kind: ServiceTelemetry
metadata:
  name: default
  namespace: service-telemetry
spec:
  alerting:
    alertmanager:
      storage:
        persistent:
          pvcStorageRequest: 20G
          storageSelector: {}
        receivers:
          snmpTraps:
            enabled: false
            target: 192.168.24.254
        strategy: ephemeral
    enabled: true
  backends:
    events:
      elasticsearch:
        enabled: false
        storage:
          persistent:
            pvcStorageRequest: 20Gi
            storageSelector: {}
          strategy: ephemeral
    metrics:
      prometheus:
        enabled: true
        scrapeInterval: 10s
        storage:
          persistent:
            pvcStorageRequest: 20G
            storageSelector: {}
          retention: 7d
          strategy: ephemeral
  graphing:
    enabled: false
    grafana:
      adminPassword: secret
      adminUser: root
      disableSignoutMenu: false
      ingressEnabled: false
  highAvailability:
    enabled: false
  transports:
    qdr:
      enabled: true
      web:
        enabled: false
```

Checking for `retention: 7d`

```
$ oc get prometheus default -oyaml

apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  creationTimestamp: "2021-01-13T21:06:16Z"
  generation: 1
  labels:
    prometheus: default
  name: default
  namespace: service-telemetry
spec:
  alerting:
    alertmanagers:
    - name: alertmanager-operated
      namespace: service-telemetry
      port: web
  replicas: 1
  retention: 7d
  ruleSelector: {}
  securityContext: {}
  serviceAccountName: prometheus-k8s
  serviceMonitorSelector:
    matchLabels:
      app: smart-gateway
```

Checking for `--storage.tsdb.retention.time=7d`

```
$ oc get statefulset.apps/prometheus-default -oyaml

apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: prometheus-default
  namespace: service-telemetry
spec:
  podManagementPolicy: Parallel
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: prometheus
      prometheus: default
  serviceName: prometheus-operated
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: prometheus
        prometheus: default
    spec:
      containers:
      - args:
        - --web.console.templates=/etc/prometheus/consoles
        - --web.console.libraries=/etc/prometheus/console_libraries
        - --config.file=/etc/prometheus/config_out/prometheus.env.yaml
        - --storage.tsdb.path=/prometheus
        - --storage.tsdb.retention.time=7d
        - --web.enable-lifecycle
        - --storage.tsdb.no-lockfile
        - --web.route-prefix=/
        image: quay.io/prometheus/prometheus:v2.7.1
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 6
          httpGet:
            path: /-/healthy
            port: web
            scheme: HTTP
          periodSeconds: 5
          successThreshold: 1
          timeoutSeconds: 3
        name: prometheus
        ports:
        - containerPort: 9090
          name: web
          protocol: TCP
        readinessProbe:
          failureThreshold: 120
          httpGet:
            path: /-/ready
            port: web
            scheme: HTTP
          periodSeconds: 5
          successThreshold: 1
          timeoutSeconds: 3
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/prometheus/config_out
          name: config-out
          readOnly: true
        - mountPath: /prometheus
          name: prometheus-default-db
        - mountPath: /etc/prometheus/rules/prometheus-default-rulefiles-0
          name: prometheus-default-rulefiles-0
<snip>
```

**Comment by leifmadsen** [comment]
Create View: Available options in the UI

image

Create View: default value is 24h

image

Status View: value is shown on screen

image

**Review by pleimer — APPROVED** [review]

File: deploy/crds/infra.watch_servicetelemetrys_crd.yaml [hunk 1]
```diff
@@ -96,6 +96,12 @@ spec:
                             enum:
                               - ephemeral
                               - persistent
+                          retention:
+                            description: Time duration Prometheus shall retain data for. Default
+                              is '24h', and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
+                              (milliseconds seconds minutes hours days weeks years).
```
Comment (csibbitt) [inline deploy/crds/infra.watch_servicetelemetrys_crd.yaml:102]: Never seen retention times in milliseconds before, but sure! Lol

**Review by csibbitt — APPROVED** [review]

File: deploy/crds/infra.watch_servicetelemetrys_crd.yaml (hunk 1 above)
Comment (leifmadsen) [reply deploy/crds/infra.watch_servicetelemetrys_crd.yaml:102]: lol yea, but it's valid based on what can be passed to the Prometheus Operator. I just copy/pasted from the Prometheus CRD :D

**Pull request merged**